src/inference/dev_api/openvino/runtime/iremote_tensor.hpp

```diff
@@ -33,5 +33,17 @@ class OPENVINO_RUNTIME_API IRemoteTensor : public ITensor {
      * @return Device name
      */
     virtual const std::string& get_device_name() const = 0;
+
+    void copy_to(const std::shared_ptr<ov::ITensor>& dst) const override {
+        copy_to(dst, 0, 0, this->get_shape());
+    }
+
+    virtual void copy_from(const std::shared_ptr<const ov::ITensor>& src) {
+        copy_from(src, 0, 0, this->get_shape());
+    }
+
+    virtual void copy_to(const std::shared_ptr<ov::ITensor>& dst, size_t src_offset, size_t dst_offset, ov::Shape roi_shape) const = 0;
+
+    virtual void copy_from(const std::shared_ptr<const ov::ITensor>& src, size_t src_offset, size_t dst_offset, ov::Shape roi_shape) = 0;
```

Comment:
why do we need such overloads? I supposed ROI can be stored within RoiTensor and copy_[to|from](tensor) can be used instead

Reply:
Hmm, do you mean that ROI shape is presented in RoiTensor itself? If so, then we need to pass somehow this information to RemoteTensor implementation, because RoiTensor just calls copy_to()/copy_from() of underlying RemoteTensor object (and this RemoteTensor doesn't know about ROI):
https://github.com/sshlyapn/openvino/blob/bb8bbd6b35f5021f5ace6933d21f0745e2f5b4cc/src/inference/src/dev/make_tensor.cpp#L358-L367
So it looks like we either need to pass offsets and ROI shape as an arguments, or introduce some ROI related fields in RemoteTensor implementation? In case of host tensors, RoiTensors just overrides data() and get_shape() functions, keeping the original copying logic. But the situation seems a little different for remote tensors, doesn't it?

Reply:
you are right